### PR TITLE
Barbarian Subclass Path of the Spirits initial implementation

### PR DESCRIPTION
**Pull request by tivie**
Initial implementation of Barbarian Subclass Path of the Spirits

### 3rd Level

- **Spirit Seeker**: Yours is a path that seeks attunement with the natural world, giving you a kinship with beasts. You gain the ability to cast the AnimalFriendship and FindTraps spells at will.
- **Animal Spirit**: At 3rd level, when you adopt this path, you choose **one** animal spirit as a guide and gain its feature.
    - **Bear**: While raging, you have resistance to all damage except psychic damage. The spirit of the bear makes you tough enough to stand up to any punishment.
    - **Eagle**: (rogue's cunning action) The spirit of the eagle makes you into a nimble predator who can weave through the fray with ease. You can take the Dash, Disengage, or Hide action as a Bonus Action.
    - **Wolf**: The spirit of the wolf makes you a leader of hunters. While you're raging, your friends have *advantage* on melee attack rolls against any creature within 5 feet of you that is hostile to you.

### 6th Level
- **Animal Aspect**: At 6th level, you gain a magical aspect (benefit) based on the spirit animal of your choice. You can choose the same animal you selected at 3rd level or a different one.
    - Bear: You gain the might of a bear. Your HP increases by 2 points for every level you take in this class and have *advantage* on Strength Checks.
    - Eagle: You gain the eyesight of an eagle. You gain superior darkvision and Keen Sight.
    - Wolf: You gain the hunting sensibilities of a wolf. You can track other creatures by smell and hearing (Keen Smell and Keen Hearing). You also gain the ability to cast the IdentifyCreatures spell at will.

### 10th Level
 - **Spirit Walker**: At 10th level, as a Bonus Action, you can summon the protection of your Animal Spirit to guide and protect you in a 15ft sphere area around you for 10 minutes.

# Issues:

 - Animal Spirit -> Wolf: Currently the effect is not working properly. It gives PackTactics to the caster while it should give it to adjacent allies. Couldn't make it work.

### CODE REVIEW

File: SolastaUnfinishedBusiness/Subclasses/PathOfTheSpirits.cs [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+
+    internal PathOfTheSpirits()
+    {
+        #region 3rd LEVEL FEATURES
+
+        // Spirit Seeker
+        // Yours is a path that seeks attunement with the natural world, giving you a kinship with beasts.
+        // At 3rd level when you adopt this path, you gain the ability to cast the AnimalFriendship and FindTraps spells at will.
+        var featureSetPathOfTheSpiritsSpiritSeeker = FeatureDefinitionFeatureSetBuilder
+            .Create("PathOfTheSpiritsSpiritSeeker")
```
Comment (ThyWoof) [inline SolastaUnfinishedBusiness/Subclasses/PathOfTheSpirits.cs:26]: prefix name with FeatureSet.

File: SolastaUnfinishedBusiness/Subclasses/PathOfTheSpirits.cs [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+                BuildSpiritSeekerSpell(SpellDefinitions.AnimalFriendship),
+                BuildSpiritSeekerSpell(SpellDefinitions.FindTraps)
+            )
+            .AddToDB();
+
+
+        // Animal Spirit
+        // At 3rd level, when you adopt this path, you choose an animal spirit as a guide and gain its feature.
+        var featureSetPathOfTheSpiritsAnimalSpirit = FeatureDefinitionFeatureSetBuilder
+        .Create(FeatureDefinitionFeatureSets.FeatureSetSorcererDraconicChoice, "PathOfTheSpiritsAnimalSpiritChoices")
```
Comment (ThyWoof) [inline SolastaUnfinishedBusiness/Subclasses/PathOfTheSpirits.cs:38]: prefix name with FeatureSet.

File: SolastaUnfinishedBusiness/Subclasses/PathOfTheSpirits.cs [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+
+        #endregion
+
+
+        #region 6th LEVEL FEATURES
+
+        // Animal Aspect
+        // At 6th level, you gain a magical aspect (benefit) based on the spirit animal of your choice. You can choose the same animal you selected at 3rd level or a different one.
+        var featureSetPathOfTheSpiritsAnimalAspect = FeatureDefinitionFeatureSetBuilder
+            .Create(FeatureDefinitionFeatureSets.FeatureSetSorcererDraconicChoice, "PathOfTheSpiritsAnimalAspectChoices")
```
Comment (ThyWoof) [inline SolastaUnfinishedBusiness/Subclasses/PathOfTheSpirits.cs:59]: prefix name with FeatureSet.

File: SolastaUnfinishedBusiness/Subclasses/PathOfTheSpirits.cs [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+
+        // Spirit Walker
+        // At 10th level, as a Bonus Action, you can summon the protection of your Animal Spirit to guide and protect you in a 15ft sphere area around you for 10 minutes.
+        // Enemy creatures in this sphere are affected by this feature. An affected creature's speed is halved in the area, and when the creature enters
+        // the area for the first time on a turn or starts its turn there, it must make a Wisdom saving throw.
+        // On a failed save, the creature takes 3d8 radiant damage (if you are good or neutral) or 3d8 necrotic damage (if you are evil).
+        // On a successful save, the creature takes half as much damage.
+        // You can use this feature 3 times per long rest.
+        var featureSetPathOfTheSpiritsSpiritWalker = FeatureDefinitionFeatureSetBuilder
+            .Create("PathOfTheSpiritsSpiritWalker")
```
Comment (ThyWoof) [inline SolastaUnfinishedBusiness/Subclasses/PathOfTheSpirits.cs:84]: prefix name with FeatureSet.

File: SolastaUnfinishedBusiness/Subclasses/PathOfTheSpirits.cs [hunk 5]
```diff
@@ -0,0 +1,240 @@
...
+        .AddFeaturesAtLevel(6,
+            featureSetPathOfTheSpiritsAnimalAspect)
+        .AddFeaturesAtLevel(10,
+            featureSetPathOfTheSpiritsSpiritWalker)
+        .AddToDB();
+    }
+    private static FeatureDefinition BuildSpiritSeekerSpell(SpellDefinition spellDefinition)
+    {
+        return FeatureDefinitionPowerBuilder
+            .Create($"PathOfTheSpirits{spellDefinition.name}")
```
Comment (ThyWoof) [inline SolastaUnfinishedBusiness/Subclasses/PathOfTheSpirits.cs:108]: prefix name with Power.

File: SolastaUnfinishedBusiness/Subclasses/PathOfTheSpirits.cs [hunk 6]
```diff
@@ -0,0 +1,240 @@
...
+            .SetGuiPresentation(spellDefinition.GuiPresentation)
+            .SetUsesFixed(ActivationTime.Action, RechargeRate.AtWill)
+            .SetEffectDescription(spellDefinition.EffectDescription)
+            .AddToDB();
+    }
+    private static FeatureDefinitionFeatureSet BuildAnimalSpiritChoice(string name, params FeatureDefinition[] featureDefinitions)
+    {
+        return FeatureDefinitionFeatureSetBuilder
```
Comment (ThyWoof) [inline SolastaUnfinishedBusiness/Subclasses/PathOfTheSpirits.cs:116]: prefix name with FeatureSet.

File: SolastaUnfinishedBusiness/Subclasses/PathOfTheSpirits.cs [hunk 7]
```diff
@@ -0,0 +1,240 @@
...
+            .Create($"PathOfTheSpiritsAnimalSpiritChoice{name}")
+            .SetGuiPresentation(Category.Feature)
+            .AddFeatureSet(featureDefinitions)
+            .AddToDB();
+    }
+    private static FeatureDefinitionFeatureSet BuildAnimalAspectChoice(string name, params FeatureDefinition[] featureDefinitions)
+    {
+        return FeatureDefinitionFeatureSetBuilder
+            .Create($"PathOfTheSpiritsAnimalAspectChoice{name}")
```
Comment (ThyWoof) [inline SolastaUnfinishedBusiness/Subclasses/PathOfTheSpirits.cs:125]: prefix name with FeatureSet.

File: SolastaUnfinishedBusiness/Subclasses/PathOfTheSpirits.cs [hunk 8]
```diff
@@ -0,0 +1,240 @@
...
+                        .AddToDB(), ConditionForm.ConditionOperation.Add)
+                    .Build())
+                .Build())
+            .AddToDB();
+    }
+
+    private static FeatureDefinition PowerBearDurability()
+    {
+        return FeatureDefinitionAttributeModifierBuilder
+            .Create("PathOfTheSpiritsBearDurability")
```
Comment (ThyWoof) [inline SolastaUnfinishedBusiness/Subclasses/PathOfTheSpirits.cs:168]: prefix name with Power.

File: SolastaUnfinishedBusiness/Subclasses/PathOfTheSpirits.cs [hunk 9]
```diff
@@ -0,0 +1,240 @@
...
+            .Create()
+            .SetBonusMode(AddBonusMode.None)
+            .SetLevelAdvancement(EffectForm.LevelApplianceType.No, LevelSourceType.ClassLevel)
+            .CreatedByCharacter()
+            .SetConditionForm(wolfLeadershipCondition, ConditionForm.ConditionOperation.Add)
+            .Build();
+            
+
+        return FeatureDefinitionPowerBuilder
+            .Create("PathOfTheSpiritsWolfLeadershipAura")
```
Comment (ThyWoof) [inline SolastaUnfinishedBusiness/Subclasses/PathOfTheSpirits.cs:194]: prefix name with Power.

File: SolastaUnfinishedBusiness/Subclasses/PathOfTheSpirits.cs [hunk 10]
```diff
@@ -0,0 +1,240 @@
...
+                .SetRecurrentEffect(RecurrentEffect.OnActivation | RecurrentEffect.OnEnter | RecurrentEffect.OnTurnStart)
+                .SetEffectForms(wolfLeadershipEffect)
+                .Build())
+            .AddToDB();
+    }
+#endif
+    private static FeatureDefinition PowerBearMight()
+    {
+        return FeatureDefinitionAbilityCheckAffinityBuilder
+            .Create("PathOfTheSpiritsBearMight")
```
Comment (ThyWoof) [inline SolastaUnfinishedBusiness/Subclasses/PathOfTheSpirits.cs:212]: prefix name with Power.

File: SolastaUnfinishedBusiness/Subclasses/PathOfTheSpirits.cs [hunk 11]
```diff
@@ -0,0 +1,240 @@
...
+            .Create("PathOfTheSpiritsWolfKnowledge")
+            .SetGuiPresentation(Category.Feature)
+            .AddFeatureSet(BuildSpiritSeekerSpell(SpellDefinitions.IdentifyCreatures))
+            .AddToDB();
+    }
+
+    private static FeatureDefinition PowerSpiritGuardian()
+    {
+        return FeatureDefinitionPowerBuilder
+            .Create("PathOfTheSpiritsSpiritGuardians")
```
Comment (ThyWoof) [inline SolastaUnfinishedBusiness/Subclasses/PathOfTheSpirits.cs:230]: prefix name with Power.

File: SolastaUnfinishedBusiness/Subclasses/PathOfTheSpirits.cs [hunk 12]
```diff
@@ -0,0 +1,240 @@
...
+            .Create("PathOfTheSpiritsBearMight")
+            .SetGuiPresentation(Category.Feature)
+            .BuildAndSetAffinityGroups(CharacterAbilityCheckAffinity.Advantage, DieType.D1, 0, (AttributeDefinitions.Strength, string.Empty))
+            .AddToDB();
+    }
+
+    private static FeatureDefinition PowerWolfKnowledge()
+    {
+        return FeatureDefinitionFeatureSetBuilder
+            .Create("PathOfTheSpiritsWolfKnowledge")
```
Comment (ThyWoof) [inline SolastaUnfinishedBusiness/Subclasses/PathOfTheSpirits.cs:221]: prefix name with FeatureSet.

**Review by ThyWoof — CHANGES_REQUESTED** [review]
Obrigado!

**Comment by ThyWoof** [comment]
Tactician's shared vigilance uses aura to give attribute bonus to initiative to allies in radius but it also gives to caster. That could be an alternative to Animal Spirit -> Wolf.

**Comment by tivie** [comment]
Feature Definition naming should be fixed now.

**Comment by tivie** [comment]
> Tactician's shared vigilance uses aura to give attribute bonus to initiative to allies in radius but it also gives to caster. That could be an alternative to Animal Spirit -> Wolf. Start there and we can write a patch later to exclude caster.

Thanks. WIll give it a try and let you know

**Review by ThyWoof — APPROVED** [review]

**Pull request merged**